File: backend/Data/DB_Models/Worker.cs
﻿using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Data.DB_Models
{
    public class Worker : Entity //Name and ID from Entity
    {
        public string Assignment { get; set; }
        [DataType(DataType.PhoneNumber)]
        public string PhoneNumber { get; set; }
        public string UserName { get; set; }
        public string Password { get; set; }
        public string Supervisor { get; set; } //not sure if string just yet
        public long DepartmentId { get; set; }
        [JsonIgnore]
        public virtual Department Department { get; set; }
    }
}

File: frontend/src/app/worker.ts
export class Worker {
    public active : boolean;
    public name : string;
    public assignment : string;
    public phoneNumber : string;
    public userName : string;
    public password : string;
    public supervisor : string;
    public id : number;
    public departmentName : string;
    public abreviation : string;
    public departmentId : number;

    constructor(active : boolean, name : string, 
        phoneNumber : string, userName : string, password : string,
         supervisor : string, assignment:string){
        this.active = active;
        this.name = name;
        this.assignment = assignment;
        this.phoneNumber = phoneNumber;
        this.userName = userName;
        this.password = password;
        this.supervisor = supervisor;
        this.id = 0;
        this.departmentId = 0;
        this.abreviation = '';
        this.departmentName = '';
    }
}
